.github/workflows/ports_esp32.yml

```diff
@@ -18,11 +23,36 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  build_idf50:
+  build_idf:
+    strategy:
+      fail-fast: false
+      matrix:
+        ci_func:  # names are functions in ci.sh
+          - esp32_s2_build
+          - esp32_s3_c3_build
```

Comment:
I would call them `esp32_build_group1` and `esp32_build_group2` (or set1/set2 or just 1/2), so their contents can change without having to change their name.  The only aim is to make them take about the same time to build.  Then we can add another easily, if/when needed.

Reply:
@dpgeorge Are you sure about this? The `build` functions in `ci.sh` currently have descriptive names (including `ci_stm32_pyb_build` and `ci_stm32_nucleo_build`, I didn't add those names in this PR).

I think that's valuable as an indication of what a particular function will do - some of the build functions are otherwise not particularly self-documenting at first glance.

If we find the name keeps getting longer as we add boards or targets to it, this is also a pretty clear sign that maybe a new job should be added instead.

Reply:
> I think that's valuable as an indication of what a particular function will do - some of the build functions are otherwise not particularly self-documenting at first glance.

Yes that's true.

Then I suggest to call them `esp32_build_s3_c3`, ie put "build" first.  Because they are subsets of the build process.